odoo/addons/base/models/ir_module.py

```diff
@@ -489,6 +489,7 @@ def module_uninstall(self):
         self.env['ir.model.data']._module_data_uninstall(modules_to_remove)
         # we deactivate prefetching to not try to read a column that has been deleted
         self.with_context(prefetch_fields=False).write({'state': 'uninstalled', 'latest_version': False})
```

Comment:
Read within the diff of Strategy2

we can `self.env.flush_all()` before this line,
so the performance impact of `self.with_context(prefetch_fields=False).env.flush_all()` can be minimized

But the strategy is not promised since `.with_context(prefetch_fields=False)` should be required for all ORM CRUD after the field columns are dropped, and we may have to add more patches if some methods are overridden.